inference/core/workflows/core_steps/transformations/dynamic_zones/v1.py

```diff
@@ -137,9 +165,24 @@ def run(
                 simplified_polygon = calculate_simplified_polygon(
                     mask=mask,
                     required_number_of_vertices=required_number_of_vertices,
+                    force_rectangle=force_rectangle,
                 )
-                if len(simplified_polygon) != required_number_of_vertices:
-                    continue
-                simplified_polygons.append(simplified_polygon)
+                if force_rectangle and required_number_of_vertices == 4:
```

Comment:
I'd vote to raise exception at the beginning of `run` method if `force_rectangle` is true and requested number of vertices was not set to 4.

Reply:
Got it. Thanks for flagging this @grzegorz-roboflow. I'll move the logic to the `run` function.